src/v/redpanda/admin_server.cc

```diff
@@ -442,3 +444,37 @@ void admin_server::register_broker_routes() {
             std::move(res));
       });
 }
+
+void admin_server::register_partition_routes() {
+    /*
+     * Get a list of partition summaries.
+     */
+    ss::httpd::partition_json::get_partitions.set(
+      _server._routes, [this](std::unique_ptr<ss::httpd::request>) {
+          using summary = ss::httpd::partition_json::partition_summary;
+          return _partition_manager
+            .map_reduce0(
+              [](cluster::partition_manager& pm) {
+                  std::vector<summary> partitions;
+                  partitions.reserve(pm.partitions().size());
+                  for (const auto& it : pm.partitions()) {
+                      summary p;
+                      p.ns = it.first.ns;
+                      p.topic = it.first.tp.topic;
+                      p.partition_id = it.first.tp.partition;
+                      p.core = ss::this_shard_id();
+                      partitions.push_back(std::move(p));
+                  }
+                  return partitions;
+              },
+              std::vector<summary>{},
+              [](std::vector<summary> acc, std::vector<summary> update) {
+                  acc.insert(acc.end(), update.begin(), update.end());
```

Comment:
These could be `move_iterators`

Reply:
https://github.com/vectorizedio/redpanda/issues/1407

thanks @BenPope this was merged but ill follow up with this change.